### PR TITLE
Added more API support

### PR DESCRIPTION
**Pull request by bartoszhernas**
Hi,

I have forked your library and added some more api support, basically everything that I had been missing for my implementation.

### CODE REVIEW

File: LayerClient/LayerClient.py [hunk 1]
```diff
@@ -143,8 +152,11 @@ def _raw_request(self, method, url, data=None, extra_headers=None):
         )
 
         if result.ok:
-            return result.json()
-
+            try:
+                return result.json()
+            except JSONDecodeError:
```
Comment (rschlaikjer) [inline LayerClient/LayerClient.py]: Doesn't `result.json()` throw a ValueError here? I think you should be able to just catch that, which saves us adding another dependency

File: LayerClient/LayerClient.py [hunk 2]
```diff
@@ -88,18 +95,20 @@ def __init__(self, app_uuid, bearer_token):
...
-    def _get_layer_headers(self):
+    def _get_layer_headers(self, method):
         """
         Convenience method for retrieving the default set of authenticated
         headers.
 
         Return: The headers required to authorize ourselves with the Layer
         platform API.
         """
+        content_type = 'application/json' if method != METHOD_PATCH else 'application/vnd.layer-patch+json'
```
Comment (rschlaikjer) [inline LayerClient/LayerClient.py]: Can you wrap this line to keep it < 80 chars?

File: LayerClient/LayerClient.py [hunk 3]
```diff
@@ -394,9 +487,17 @@ class Sender:
     one over the other.
     """
 
-    def __init__(self, id=None, name=None):
+    def __init__(self, id=None, name=None, display_name=None, avatar_url=None, first_name=None, last_name=None, phone_number=None, email_address=None, metadata=None):
```
Comment (rschlaikjer) [inline LayerClient/LayerClient.py]: This line is also a bit long if you wouldn't mind wrapping it

**Comment by rschlaikjer** [comment]
Thanks for the feature PR Bartosz! Just got a couple small comments, but if you address those I'll be happy to merge.

File: LayerClient/LayerClient.py (hunk 1 above)
Comment (bartoszhernas) [inline LayerClient/LayerClient.py]: You are right, `JSONDecodeError` is a subclass of `ValueError`

**Comment by bartoszhernas** [comment]
Cool! Fixed all your comments.

I have basically run PyCharm's defaults Python code formatter with 80 chars code width.
It changed few more things, if you don't like them, I can revert it.

**Comment by rschlaikjer** [comment]
Nice, formatting looks good. 

Looks like there's three test failures around announcements / messages - mind taking a look at those?

**Comment by bartoszhernas** [comment]
Sure, will do 👍

**Comment by bartoszhernas** [comment]
`14 passed in 0.17 seconds`

All fixed

**Comment by rschlaikjer** [comment]
Fantastic! Just merged someone else's PR around read receipts, looks like you've got some merge conflicts. Once you pull master I'll merge this and cut a release to pypi

**Comment by bartoszhernas** [comment]
Done :)

**Comment by rschlaikjer** [comment]
Merged! I've just released version 0.1.7 which contains your changes :)
https://pypi.python.org/pypi/LayerClient/0.1.7

**Pull request merged**